alica_dynamic_loading/src/DynamicBehaviourCreator.cpp

```diff
@@ -19,20 +19,28 @@ DynamicBehaviourCreator::DynamicBehaviourCreator()
 
 std::unique_ptr<BasicBehaviour> DynamicBehaviourCreator::createBehaviour(int64_t behaviourId, BehaviourContext& context)
 {
-    std::string completeLibraryName = calculateLibraryCompleteName(_libraryPath, context.behaviourModel->getLibraryName());
-    if (completeLibraryName.empty()) {
-        Logging::logError("DynamicLoading") << "Could not compute the complete library name for creating the behaviour: " << context.name;
-        return nullptr;
+    std::string completeLibraryName;
+    try {
+        completeLibraryName = calculateLibraryCompleteName(_libraryPath, context.behaviourModel->getLibraryName());
+    } catch (const DynamicLoadingException& ex) {
+        throw DynamicLoadingException{
+                "behaviour", behaviourId, context.name, context.behaviourModel->getImplementationName(), context.behaviourModel->getLibraryName(), ex.what()};
     }
 
-    _behaviourCreator = boost::dll::import_alias<behaviourCreatorType>( // type of imported symbol must be explicitly specified
-            completeLibraryName,                                        // complete path to library also with file name
-            context.behaviourModel->getImplementationName(),            // symbol to import
-            boost::dll::load_mode::append_decorations                   // do append extensions and prefixes
-    );
-    Logging::logDebug("DynamicLoading") << "Loaded behavior " << context.behaviourModel->getName();
+    try {
+        _behaviourCreator = boost::dll::import_alias<behaviourCreatorType>( // type of imported symbol must be explicitly specified
+                completeLibraryName,                                        // complete path to library also with file name
+                context.behaviourModel->getImplementationName(),            // symbol to import
+                boost::dll::load_mode::append_decorations                   // do append extensions and prefixes
+        );
+    } catch (const std::exception& ex) {
+        // import_alias can throw boost::system::system_error or std::bad_alloc, this block handles both since each of them derives from std::exception
+        throw DynamicLoadingException{
+                "behaviour", behaviourId, context.name, context.behaviourModel->getImplementationName(), context.behaviourModel->getLibraryName(), ex.what()};
+    }
 
     std::unique_ptr<BasicBehaviour> createdBehaviour = _behaviourCreator(context);
+    Logging::logDebug("DynamicLoading") << "Loaded behavior " << context.behaviourModel->getName();
```

Comment:
I suppose name is better because we can always easily fetch the implementation name from the name but the reverse is not necessarily possible if more than one behaviour uses the same implementation name. Not a major problem, but name is easier I think.